src/server/services/wms/qgswmsgetlegendgraphics.cpp

```diff
@@ -38,37 +38,37 @@ namespace QgsWms
 
     QgsWmsParameters wmsParameters( QUrlQuery( request.url() ) );
 
+    ImageOutputFormat outputFormat = parseImageFormat( format );
+
+    QString saveFormat;
+    QString contentType;
+    switch ( outputFormat )
+    {
+      case PNG:
+      case PNG8:
+      case PNG16:
+      case PNG1:
+        contentType = "image/png";
+        saveFormat = "PNG";
+        break;
+      case JPEG:
+        contentType = "image/jpeg";
+        saveFormat = "JPEG";
+        break;
+      default:
+        throw QgsServiceException( "InvalidFormat",
+                                   QString( "Output format '%1' is not supported in the GetLegendGraphic request" ).arg( format ) );
```

Comment:
Unrelated suggestion

```suggestion
                                   QStringLiteral( "Output format '%1' is not supported in the GetLegendGraphic request" ).arg( format ) );
```